### ISSUE TITLE
type for `$push` is incorrect.

### ISSUE THREAD

**Issue by imranbarbhuiya** [Automattic/mongoose#12126]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

^6.4.5

### Node.js version

v18

### MongoDB server version

5.0

### Description

Type for `$push` is incorrect.

### Steps to Reproduce

1. Create a model
```ts
const categorySchema = new Schema({
	name: String
});

const mainSchema = new Schema({
	name: String,
	categories: [categorySchema]
});

const mainModel = new model(mainSchema, 'mainSchema');
```

2. Try to `$push` to the `categories` field.
```ts
await mainModel.updateOne(
	{ name: 'name' },
	{
		$push: {
			categories: {
				name: 'ok'
			}
		}
	}
);
```

**Comment by imranbarbhuiya** [comment]
Also 3rd issue for https://github.com/Automattic/mongoose/issues/12030 is only fixed when anyone uses `new Schema` but it's not fixed when passing like this
```ts
const userSchema = new Schema({
	users: [
		{
			username: { type: String },
			credit: { type: Number, default: 0 }
		}
	]
});

const userModel = model('user', userSchema);

const data = await userModel.findOne();
data!.users.push({ username: 'parbez' });
// ts warns to add credit which can be so
```

**Comment by vkarpov15** [comment]
Fixed in 6.5.1 with b57efecf17648390d5b576a4bb704c0f0f27d309
